[compiler-rt][test] Rewrote test to remove curly braces

connieyzhu opened the pull request:
This patch removes curly braces from a test, as lit's internal shell implementation does not support curly brace syntax.

Fixes https://github.com/llvm/llvm-project/issues/102382.

connieyzhu committed:
[compiler-rt][test] Rewrote test to remove curly braces

This patch removes curly braces from a test, as lit's internal shell
implementation does not support curly brace syntax.

ilovepi commented on compiler-rt/test/asan/TestCases/Linux/dlopen-mixed-c-cxx.c:
```diff
@@ -1,7 +1,7 @@
 // RUN: %clang_asan %s -o %t.out -ldl
 //
-// RUN: { env ASAN_OPTIONS=verbosity=1 %t.out %t.so || : ; } 2>&1 | FileCheck %s
+// RUN: env ASAN_OPTIONS=verbosity=1 %t.out %t.so 2>&1 | FileCheck %s || : 
```
I don't think these are equivalent, and we probably don't want to change the return code for `FileCheck`, since we still want CHECK lines to trigger a failure. It seems that this program will always return a non-zero exit code, so maybe we can just use `not` instead of `|| :`? Alternatively, we could make the program return `0`, but I'm guessing the author were worried about some optimization preventing the test from working correctly, and I'm not familar enough w/ this test to know if that would be problematic.

connieyzhu committed:
[compiler-rt][test] Fix incorrect logic with using '|| :' in test

This patch fixes incorrect usage of '|| :', which modified the function
of the original test. '|| :' should be used with the env command, and
not with FileCheck.

arichardson commented on compiler-rt/test/asan/TestCases/Linux/dlopen-mixed-c-cxx.c:
```diff
@@ -1,7 +1,10 @@
-// RUN: { env ASAN_OPTIONS=verbosity=1 %t.out %t.so || : ; } 2>&1 | FileCheck %s
+// The program can potentially return a non-zero exit code, so use || : to
+// ensure command returns true
+// RUN: env ASAN_OPTIONS=verbosity=1 %t.out %t.so &> %t_env || :
```
The initial version of this commit used `not`: https://github.com/llvm/llvm-project/commit/cbd28cd05bb11e9d76d71c6cc2d38d89dbb1fe1a but for some reason that was changed in https://github.com/llvm/llvm-project/commit/a30a4a35ecbd2ebff79de8a193e7045b17da3ff5 

Maybe removing `not` was used to work around the missing `UNSUPPORTED:` that was only added in https://github.com/llvm/llvm-project/commit/307daa71a8f0bb258d4c89d5bad2579f468162ca CC: @serge-sans-paille @hahnjo 

I'd be tempted to restore this to using `not` instead of `|| :` and see if the builders are happy.

connieyzhu replied:
I replaced `||:` with `not` in my most recent commit. The tests are passing when I run things locally - still need to wait for builders though.

connieyzhu committed:
[compiler-rt][test] Replaced ||: with not

arichardson commented on compiler-rt/test/asan/TestCases/Linux/dlopen-mixed-c-cxx.c:
```diff
@@ -1,7 +1,9 @@
 // RUN: %clang_asan %s -o %t.out -ldl
 //
-// RUN: { env ASAN_OPTIONS=verbosity=1 %t.out %t.so || : ; } 2>&1 | FileCheck %s
+// The program can potentially return a non-zero exit code, so use || : to
```
Stale comment here.

arichardson reviewed (approved):
Comment needs updating otherwise LGTM

connieyzhu committed:
[NFC] Removed stale comment
